project/apps/questions-service/src/questions.controller.ts

```diff
@@ -2,19 +2,43 @@ import { Controller, UsePipes } from '@nestjs/common';
 import { MessagePattern, Payload } from '@nestjs/microservices';
 import { QuestionsService } from './questions.service';
 import { ZodValidationPipe } from '@repo/pipes/zod-validation-pipe.pipe';
-import { CreateQuestionDto, createQuestionSchema } from '@repo/dtos/questions';
+import {
+  CreateQuestionDto,
+  createQuestionSchema,
+  UpdateQuestionDto,
+  updateQuestionSchema,
+} from '@repo/dtos/questions';
 @Controller()
 export class QuestionsController {
   constructor(private readonly questionsService: QuestionsService) {}
 
+  @MessagePattern({ cmd: 'get_questions' })
+  async getQuestions() {
+    return await this.questionsService.findAll();
+  }
+
   @MessagePattern({ cmd: 'get_question' })
-  getQuestion(id: string) {
-    return this.questionsService.findById(id);
+  async getQuestionById(id: bigint) {
+    return await this.questionsService.findById(id);
   }
 
   @MessagePattern({ cmd: 'create_question' })
   @UsePipes(new ZodValidationPipe(createQuestionSchema))
-  async create(@Payload() createQuestionDto: CreateQuestionDto) {
-    return createQuestionDto;
+  async createQuestion(@Payload() createQuestionDto: CreateQuestionDto) {
+    return await this.questionsService.create(createQuestionDto);
+  }
+
+  @MessagePattern({ cmd: 'update_question' })
+  @UsePipes(new ZodValidationPipe(updateQuestionSchema))
+  async updateQuestion(@Payload() createQuestionDto: UpdateQuestionDto) {
+    return await this.questionsService.update(createQuestionDto);
+  }
+
+  // probably need to add some middleware to check if the user is an admin
+  // not sure to do it here or in the service
+  @MessagePattern({ cmd: 'delete_question' })
+  async deleteQuestion(@Payload() id: bigint) {
+    return `Not implemented: deleteQuestion(${id})`;
+    // return await this.questionsService.delete(id);
```

Comment:
@javinchua where shld we add admin privileges check

Reply:
I think we can add the admin privileges check as a separate PR because it will be q a big change but the idea is
1. We have an `auth-service` that does the checking of user in supabase, and retrieving the user information such as what role it has. (there is a few ways we can choose to store roles in user table but one way is `user_metadata`)
2. We only use this `auth-service` within our api-gateway. So this will be implemented as a `Guard` [See here for more details](https://docs.nestjs.com/guards#authorization-guard)
3. In our microservice/api-gateway where we define the endpoint (not too sure where we should put it under) We have a [Role-based authentication](https://docs.nestjs.com/guards#role-based-authentication)

Reply:
If this makes sense I can work on this^